Nodejs/Product/TestAdapter/TestExecutorWorker.cs

```diff
@@ -316,11 +318,47 @@ void RecordEnd(TestCaseResult test, ResultObject resultObject)
             }
         }
 
+        private int GetParentProcessId(uint processId, int counter = 0)
+        {
+            var query = string.Format("SELECT ParentProcessId FROM Win32_Process WHERE ProcessId = {0}", processId);
+            var search = new ManagementObjectSearcher("root\\CIMV2", query);
+            var results = search.Get().GetEnumerator();
+            results.MoveNext();
+
+            var parentId = (uint)results.Current["ParentProcessId"];
+
+            try
+            { 
+                var parentName = Process.GetProcessById((int)parentId).ProcessName;
+
+                if (parentName == "devenv")
+                    return (int)parentId;
+                if (counter < 8) //devenv.exe is usually the 4th process app, so don't search forever
+                    return GetParentProcessId(parentId, counter++);
+            }
+            catch
+            {
+                //return 0 if we fail to find devenv before we run out of parents
+                return 0;
+            }
+
+            return 0;
+        }
+
         private bool HasVisualStudioProcessId(out int processId)
         {
             processId = 0;
             var pid = Environment.GetEnvironmentVariable(NodejsConstants.NodeToolsProcessIdEnvironmentVariable);
-            return pid == null ? false : int.TryParse(pid, out processId);
+
+            if (pid != null)
+            {
+                return pid == null ? false : int.TryParse(pid, out processId);
+            }
+            else
+            {
+                processId = GetParentProcessId((uint)Process.GetCurrentProcess().Id);
```

Comment:
Could you add a comment stating the need for checking the process Id?

Reply:
I thought that including in the PR was enough, I will add the same comment I added here :)